Add golang env variables to setup documentation (#12533)
DataDog/documentation: content/en/tracing/setup_overview/setup/go.md

The Go tracer supports additional environment variables and functions for configuration.
See all available options in the [configuration documentation][8].

`DD_VERSION`
: Set the application’s version, for example: `1.2.3`, `6c44da20`, `2020.02.13`

`DD_SERVICE`
: The service name to be used for this application. 

`DD_ENV`
: Set the application’s environment, for example: prod, pre-prod, staging.

`DD_AGENT_HOST`
: **Default**: `localhost` <br>
Override the address of the trace Agent host that the default tracer attempts to submit traces to.

`DD_DOGSTATSD_PORT`
: **Default**: `8125` <br>
Override the port that the default tracer submits DogStatsD metrics to.

`DD_TAGS`
: **Default**: [] <br>

> A list of default tags to be added to every span and profile, for example: layer:api,team:intake

Reviewer: We should specify here that this list may be *either* space-separated or comma-separated.
Space-separated tags are compatible with DD_TAGS in the agent, so is recommended. Comma-separation support is present for backwards-compatibility.

Author: 👍

The Go tracer supports additional environment variables and functions for configuration.
See all available options in the [configuration documentation][8].

`DD_VERSION`
: Set the application’s version, for example: `1.2.3`, `6c44da20`, `2020.02.13`

`DD_SERVICE`
: The service name to be used for this application. 

`DD_ENV`
: Set the application’s environment, for example: prod, pre-prod, staging.

`DD_AGENT_HOST`
: **Default**: `localhost` <br>

> Override the address of the trace Agent host that the default tracer attempts to submit traces to.

Reviewer 2: ```suggestion
Override the default trace Agent host address for trace submission.
```

`DD_DOGSTATSD_PORT`
: **Default**: `8125` <br>

> Override the port that the default tracer submits DogStatsD metrics to.

Reviewer 2: ```suggestion
Override the default trace Agent port for DogStatsD metric submission.
```

`DD_TAGS`
: **Default**: [] <br>
A list of default tags to be added to every span and profile. Tags can be separated by commas or spaces, for example: `layer:api,team:intake` or `layer:api team:intake`

`DD_TRACE_STARTUP_LOGS`
: **Default**: `true` <br>

> Enable startup configuration and diagnostic log

Reviewer 2: ```suggestion
Enable startup configuration and the diagnostic log.
```
